api/controllers/calendarEntry.controller.ts

```diff
@@ -180,14 +145,73 @@ export const createCalendarEntry = async (
 };
 
 export const getCalendarEntries = async (
-  _req: Request,
+  req: Request,
   res: Response,
   _next: NextFunction,
 ) => {
   try {
-    const entries: CalendarEntry = await CalendarEntry.find();
-    res.status(200).json(entries);
+    const { start, end } = req.query;
+    const nonRecurringEntries: NonRecurringEntry[] = await CalendarEntry.find()
+      .where("recurring")
+      .equals(false)
+      .where("startTimeUtc")
+      .gte(start)
+      .where("endTimeUtc")
+      .lte(end);
+
+    const recurringEntriesWithinDateRange: RecurringEntry[] =
+      await CalendarEntry.find()
+        .where("recurring")
+        .equals(true)
+        .where("startTimeUtc")
+        .gte(start)
+        .where("recurrenceEndsUtc")
+        .lte(end);
+
+    const recurringEntriesBeginsBeforeEndsWithin: RecurringEntry[] =
+      await CalendarEntry.find()
+        .where("recurring")
+        .equals(true)
+        .where("startTimeUtc")
+        .lt(start)
+        .where("recurrenceEndsUtc")
+        .lte(end);
+
+    const recurringEntriesBeginsWithinEndsAfter: RecurringEntry[] =
+      await CalendarEntry.find()
+        .where("recurring")
+        .equals(true)
+        .where("startTimeUtc")
+        .gte(start)
+        .where("recurrenceEndsUtc")
+        .gte(end);
+
+    const recurringEntriesBeginsBeforeEndsAfter: RecurringEntry[] =
+      await CalendarEntry.find()
+        .where("recurring")
+        .equals(true)
+        .where("startTimeUtc")
+        .lt(start)
+        .where("recurrenceEndsUtc")
+        .gt(end);
+
+    const allRecurringEntries = [
+      ...recurringEntriesWithinDateRange,
+      ...recurringEntriesBeginsBeforeEndsWithin,
+      ...recurringEntriesBeginsWithinEndsAfter,
+      ...recurringEntriesBeginsBeforeEndsAfter,
+    ];
+    let allRecurrences = [];
+    allRecurringEntries.forEach((recurringEntry) => {
+      const expandedEntries = expandRecurringEntry(recurringEntry, start, end);
+      allRecurrences.push(expandedEntries);
+    });
+
+    const allEntries = [...nonRecurringEntries, ...allRecurrences.flat()];
```

Comment:
<img width="167" alt="image" src="https://user-images.githubusercontent.com/31298051/211726493-1f39a0f3-5023-4196-962b-44ba44f00cf6.png">

I think it's a bit clearer to spread the `expandedEntries` when pushing to the array. Then you don't have to call `.flat()` and the reader doesn't have to wonder and trace back the code for why this array is nested. 
```suggestion
    allRecurringEntries.forEach((recurringEntry) => {
      const expandedEntries = expandRecurringEntry(recurringEntry, start, end);
      allRecurrences.push(...expandedEntries);
    });

    const allEntries = [...nonRecurringEntries, ...allRecurrences];
```